Code/Tools/AssetProcessor/AssetBuilder/AssetBuilderComponent.cpp

```diff
@@ -370,43 +407,46 @@ bool AssetBuilderComponent::ConnectToAssetProcessor()
 
 //////////////////////////////////////////////////////////////////////////
 
-bool AssetBuilderComponent::RunInResidentMode()
+bool AssetBuilderComponent::SendRegisteredBuildersToAp()
 {
-    using namespace AssetBuilderSDK;
-    using namespace AZStd::placeholders;
+    AssetBuilderSDK::BuilderRegistrationRequest registrationRequest;
 
-    AZ_TracePrintf("AssetBuilderComponent", "RunInResidentMode: Starting resident mode (waiting for commands to arrive)\n");
-
-    AZStd::string port, id, builderFolder;
-
-    if (!GetParameter(s_paramId, id)
-        || !GetParameter(s_paramModule, builderFolder))
+    for (const auto& [uuid, desc] : m_assetBuilderDescMap)
     {
-        return false;
+        AssetBuilderSDK::BuilderRegistration registration;
+
+        registration.m_name = desc->m_name;
+        registration.m_analysisFingerprint = desc->m_analysisFingerprint;
+        registration.m_flags = desc->m_flags;
+        registration.m_flagsByJobKey = desc->m_flagsByJobKey;
+        registration.m_version = desc->m_version;
+        registration.m_busId = desc->m_busId;
+        registration.m_patterns = desc->m_patterns;
+        registration.m_productsToKeepOnFailure = desc->m_productsToKeepOnFailure;
+
+        registrationRequest.m_builders.push_back(AZStd::move(registration));
     }
 
-    if (!LoadBuilders(builderFolder))
-    {
-        return false;
-    }
+    bool result = SendRequest(registrationRequest);
```

Comment:
if the result is false, should the registrationRequest.m_builders remove this builder?

Reply:
This result is for sending the entire packet of builders over to the AP.  If this fails AssetBuilder ends up shutting down.